Languages/IronPython/IronPython/Runtime/Exceptions/TraceBack.cs

```diff
@@ -238,24 +237,14 @@ public class TraceBackFrame {
                     throw PythonOps.ValueError("lineno must be an integer");
                 }
 
-                int newLineNum = (int)value;
-
-                if (_traceAdapter != null) {
-                    SetLineNumber(newLineNum);
-                } else {
-                    throw PythonOps.ValueError("f_lineno can only be set by a trace function");
-                }
+                SetLineNumber((int)value);
             }
         }
 
         private void SetLineNumber(int newLineNum) {
             var pyThread = PythonOps.GetFunctionStackNoCreate();
-            if (!IsTopMostFrame(pyThread)) {
-                if (!TracingThisFrame(pyThread)) {
```

Comment:
This change was triggered by the stdlib test suite. One of the tests tried to modify the line number of a non-topmost stack frame and expected an exception while doing so.